loaders/src/glTF/2.0/Extensions/KHR_mesh_instancing.ts

```diff
@@ -0,0 +1,98 @@
+            return this._loader.loadNodeAsync(`#/nodes/${node.index}`, node, (babylonTransformNode) => {
+                const promises = new Array<Promise<any>>();
+                const attributeBuffers : { [name: string]: { buffer : Float32Array, accessor : IAccessor } } = {};
+                const loadAttribute = (attribute: string) => {
+                    if (extension.attributes[attribute] == undefined) {
+                        return;
+                    }
+                    const accessor = ArrayItem.Get(`${extensionContext}/attributes/${attribute}`, this._loader.gltf.accessors, extension.attributes[attribute]);
+                    promises.push(this._loader._loadFloatAccessorAsync(`/accessors/${accessor.bufferView}`, accessor).then((data) => {
+                        attributeBuffers[attribute] = { buffer : data, accessor : accessor };
+                    }));
+                };
+                const attributes = ["TRANSLATION", "ROTATION", "SCALE"];
+                for (let i = 0; i < attributes.length; ++i) {
+                    loadAttribute(attributes[i]);
+                }
+
+                return Promise.all(promises).then(() => {
+                    const instanceCount = attributeBuffers[attributes[0]].accessor.count;
+                    const digitLength = instanceCount.toString().length;
+                    const padNumber = function(num: number, length: number) {
+                        var str = String(num);
+                        while (str.length < (length)) { str = "0" + str; }
+                        return str;
+                    };
+
+                    for (let i = 0; i < instanceCount; ++i) {
+                        if (node._primitiveBabylonMeshes) {
+                            for (let j = 0; j < node._primitiveBabylonMeshes.length; ++j) {
+                                const babylonMeshPrimitive = node._primitiveBabylonMeshes[j];
+                                const instance = babylonMeshPrimitive.clone((babylonMeshPrimitive.name || babylonMeshPrimitive.id) + "_" + padNumber(i, digitLength), babylonTransformNode, true);
+
+                                if (instance) {
+                                    attributeBuffers["TRANSLATION"] ? Vector3.FromArrayToRef(attributeBuffers["TRANSLATION"].buffer, i * 3, instance.position)
+                                                                    : instance.position.set(0, 0, 0);
+                                    attributeBuffers["ROTATION"] ? Quaternion.FromArrayToRef(attributeBuffers["ROTATION"].buffer, i * 4, instance.rotationQuaternion!)
+                                                                 : instance.rotationQuaternion?.set(0, 0, 0, 1);
+                                    attributeBuffers["SCALE"] ? Vector3.FromArrayToRef(attributeBuffers["SCALE"].buffer, i * 3, instance.scaling)
+                                                              : instance.scaling.set(1, 1, 1);
```

Comment:
Once you are using createInstance, there is no need to set the position, rotationQuaternion, scaling if there is no attribute for them since they default to zero, identity, one respectively.

Reply:
I'm not too sure if that's guaranteed: https://github.com/BabylonJS/Babylon.js/blob/master/src/Meshes/instancedMesh.ts#L48-L50

In the case of our sample scene, we have two instances of the same mesh, each with their own world matrix. when we import this scene, the second teapot mesh is created as an instance.

When we create an instance of the instance's source mesh, we would have a mesh with world matrix = the source mesh, right? we should still need to set this.